constructor_exercises.js

```diff
@@ -0,0 +1,196 @@
+// //Q1
+
+// function Person(firstName, lastName, middleName) {
+//     this.firstName = firstName;
+//     this.lastName = lastName;
+//     this.middleName = middleName;
+// }
+
+// let sam = new Person("Samantha", "Jimenez", "B")
+// let erik = new Person("Erikson", "Castro", "A")
+
+// // console.log(erik.firstName);
+
+// Person.prototype.fullName = function() {console.log(this.firstName + " " + this.middleName + " " + this.lastName)}
+
+// erik.fullName();
+// sam.fullName();
+
+// //Q2
+
+// function Book(title, author, rating){
+//     this.title= title;
+//     this.author= author;
+//     this.rating= rating;
+// }
+// let book1=new Book("Phaedo", "Plato", 10);
+// let book2= new Book("Harry Potter", "J.K. Rowling", 6);
+// Book.prototype.isGood= function () {
+//     if(this.rating>=7){
+//     console.log("True");   
+// }
+// }
+// book1.isGood();
+
+//Q3.
+
+function Dog(name, breed, mood, hungry) {
+    this.name = name;
+    this.breed = breed;
+    this.mood = mood; 
+    this.hungry = hungry;
+}
+Dog.prototype.playFetch = function() {
+    this.hungry = true;
+    this.mood = "playful";
+    console.log("Ruff!");
+}
+
+let spot = new Dog("spot", "boxer", "angry", true);
+// spot.playFetch();
+// console.log(spot);
+
+Dog.prototype.feed = function() {
+    if(this.hungry === true) {
```

Comment:
Because this.hungry evaluates to a boolean there is no need to do the === true. It's not wrong, just not needed.